### PR TITLE
Implemented localization and dialogue in external files

### PR DESCRIPTION
**Pull request by Ev1lbl0w**
I've implemented localization *(and therefore dialogues in external files)* for the project. I've decided to use the PO format, as the CSV was becoming ugly and unmanageable. No work was done yet to rework the dialogue though, I've only make the dialogue read from external files.

### CODE REVIEW

File: Maps/Moki Town/HeroHome.gd [hunk 1]
```diff
@@ -37,27 +37,12 @@ func interaction(node):
...
-	"It's crammed full of books about Pokémon",
-	"\"Me and my Owten\",",
-	"\"Jerbolta's Big Adventure\",",
-	"\"Quest for the Legends\".",
-	"I've read all of these many times."
-	], false)
+	newdialog(tr("INTERACT_BOOKSHELF"), false)
 func newdialog(var text, var forceArrow):
 	dialog = dialogBox.instance()
 	dialog.loadText(text, forceArrow)
```
Comment (godofgrunts) [inline Maps/Moki Town/HeroHome.gd]: Should follow Godot's coding standards. Use `new_dialog` and `force_arrow`

File: Utilities/Dialogue Box.gd [hunk 2]
```diff
@@ -10,13 +10,13 @@ var isFinished = false
...
+	text = parse_string(textArray)
+	textLines = text.size()
 	forceArrow = forceArrowShow
 	pass
 func loadRichText(textArray, forceArrowShow):
-	text = textArray
-	textLines = textArray.size()
+	text = parse_string(textArray)
+	textLines = text.size()
 	forceArrow = forceArrowShow
```
Comment (godofgrunts) [inline Utilities/Dialogue Box.gd]: Use Godot's coding standards. `text_array` `text_lines`

File: Utilities/Dialogue Box.gd [hunk 3]
```diff
@@ -108,4 +108,11 @@ func Finished():
...
-	pass
\ No newline at end of file
+	pass
+
+func parse_string(text: String):
+	var expandedText = expand(text)
+	return expandedText.split("\n")
+
+func expand(text: String):
+	return text.format({"TrainerName": Global.TrainerName})
```
Comment (godofgrunts) [inline Utilities/Dialogue Box.gd]: Ace wants to stick to dynamic coding so please remove the static type hints.

Godot code standards again. `expanded_text`

File: Dialogue/Objects.pot [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+"It's a Pokémon battle simulation game\n"
+"called \"Red and Blue Version\"\n"
+"...Okay, time to go!"
+
+msgid "INTERACT_CONSOLE"
+msgstr "It's a Nintendo Wii U.\n"
+"But the new Nintendo Switch is better."
+
+msgid "INTERACT_BOOKSHELF"
+msgstr "It's crammed full of books about Pokémon\n"
```
[the remark this answers is not in the document]
Comment (Ev1lbl0w) [reply Dialogue/Objects.pot]: Right, I'll make them more specific

File: Maps/Moki Town/HeroHome.gd (hunk 1 above)
Comment (Ev1lbl0w) [reply Maps/Moki Town/HeroHome.gd]: Oh yeah, I forgot to apply the Godot codestly to the existing code, my bad :stuck_out_tongue:

File: Dialogue/Objects.pot (hunk 4 above)
[the remark this answers is not in the document]
Comment (Ev1lbl0w) [reply Dialogue/Objects.pot]: Fixed

File: Maps/Moki Town/HeroHome.gd (hunk 1 above)
Comment (Ev1lbl0w) [reply Maps/Moki Town/HeroHome.gd]: Fixed

Comment (Ev1lbl0w) [reply Maps/Moki Town/HeroHome.gd]: Fixed

File: Utilities/Dialogue Box.gd (hunk 2 above)
Comment (Ev1lbl0w) [reply Utilities/Dialogue Box.gd]: Fixed

File: Utilities/Dialogue Box.gd (hunk 3 above)
Comment (Ev1lbl0w) [reply Utilities/Dialogue Box.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd [hunk 5]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
+	newdialog(tr("CUTSCENE_INTRO_D3"), false)
 	dialog.rect_position = BOTTOM
 	pass
 func d4():
-	newdialog([
-	"Although they both worked demanding jobs,",
-	"they always found time to spend at home",
-	"with young " + TrainerName + "."
-	], true)
+	newdialog(tr("CUTSCENE_INTRO_D4"), true)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 6]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D2"), false)
 	dialog.rect_position = BOTTOM
 	pass
 func d3():
-	newdialog([
-	"Their mother, Lucille, was a brilliant",
-	"scientist and an expert on Nuclear Energy."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D3"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 7]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
+	newdialog(tr("CUTSCENE_INTRO_D1"), true)
 	dialog.rect_position = MIDDLE
 	pass
 	
 func d2():
-	newdialog([
-	"Their father, Kellyn, was a brave",
-	"and talented Pokémon Ranger."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D2"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 8]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
+	newdialog(tr("CUTSCENE_INTRO_D4"), true)
 	dialog.rect_position = MIDDLE
 	pass	
 func d5():
-	newdialog([
-	"The Tandor region was at peace,",
-	"and Lucille's company was working on a form",
-	"of clean energy that would change the world"
-	], true)
+	newdialog(tr("CUTSCENE_INTRO_D5"), true)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 9]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	dialog.rect_position = MIDDLE
 	pass
 func d6():
-	newdialog([
-	"The future seemed bright.",
-	"Until..."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D6"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 10]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	dialog.rect_position = MIDDLE
 	pass
 func d7():
-	newdialog([
-	"It was an ordinary day at the Nuclear",
-	"Power Plant where Lucille worked."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D7"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 11]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	dialog.rect_position = MIDDLE
 	pass
 func d8():
-	newdialog([
-	"As part of a routine inspection,",
-	"they brought in a specialist to take",
-	"a look at the electrical system."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D8"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 12]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	dialog.rect_position = MIDDLE
 	pass
 func d9():
 	stopLucilleAni()
 	$Lucille.frame = 8
-	newdialog([
-	"Cameron, have you finished your",
-	"diagnostics?"
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D9"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 13]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	pass
 func d10():
 	$Room/Cam.frame = 4
-	newdialog([
-	"Yes, I have.",
-	"Everything seems to be working perfectly.",
-	"This generator you designed really does",
-	"its job well, Lucille."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D10"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 14]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
+	newdialog(tr("CUTSCENE_INTRO_D11"), false)
 	ShowArrow($Lucille.position, BOTTOM)
 	pass
 func d12():
 	$Lucille.frame = 4
-	newdialog([
-	"If everybody has finished their work,",
-	"we'll break for lunch."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D12"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 15]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	ShowArrow($Lucille.position, BOTTOM)
 	pass
 func d13():
-	newdialog([
-	"Uh, boss...",
-	"You might want to take a look at this."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D13"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 16]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	$Lucille.frame = 12
-	newdialog([
-	"Strange... The temperature readings",
-	"are far higher than normal.",
-	"Either we need to reset the thermometer,",
-	"or...",
-	"The reactor is leaking a",
-	"massive amount of heat..."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D14"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 17]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	pass
 func d15():
 	$Room/Cam/AnimationPlayer.stop()
 	$Room/Cam.frame = 12
-	newdialog([
-	"It can't be the thermometer",
-	"I just checked that.",
-	"So..."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D15"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 18]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	ShowArrow($Room.to_global($Room/Cam.position), BOTTOM)
 	pass
 func d16():
-	newdialog([
-	"There must be a breach in the reactor!",
-	"At this temperature, in just a few minutes,",
-	"we'll have a total meltdown!"
-	], true)
+	newdialog(tr("CUTSCENE_INTRO_D16"), true)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 19]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	ShowArrow($Lucille.position, BOTTOM)
 	pass
 func d17():
-	newdialog([
-	"Everybody needs to evacuate! NOW!"
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D17"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 20]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	ShowArrow($Lucille.position, BOTTOM)
 	pass
 func d18():
-	newdialog([
-	"Lucille!",
-	"You have to leave, too!"
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D18"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

File: IntroScenes/FirstStoryCutScene.gd [hunk 21]
```diff
@@ -31,226 +31,119 @@ func _ready():
...
 	dialog.rect_position = TOP
 	ShowArrow($Room.to_global($Room/Cam.position), TOP)
 	pass
 func d19():
-	newdialog([
-	"..."
-	], false)
+	newdialog(tr("CUTSCENE_INTRO_D19"), false)
```
Comment (godofgrunts) [inline IntroScenes/FirstStoryCutScene.gd]: I think you missed some of the `newdialog`. Should be `new_dialog`

Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 20 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 19 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 18 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 17 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 16 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 15 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 14 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 13 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 12 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 11 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 10 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 9 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 8 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 7 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 6 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

File: IntroScenes/FirstStoryCutScene.gd (hunk 5 above)
Comment (Ev1lbl0w) [reply IntroScenes/FirstStoryCutScene.gd]: Fixed

**Pull request merged**